File: IsPangram/IsPangram.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
  
        Scanner sc = new Scanner(System.in);        
        String sentence = sc.nextLine(); //take input String using nextLine() as sentence may contain spaces   
        sentence = sentence.toLowerCase();
        //so that when we check for duplicates upper and lower case alphabet will be treated as same ie considered duplicate
        sentence = sentence.replaceAll("\\s+","");//for removing space
        char []pangramChk = sentence.toCharArray(); //converting string to array 
        Arrays.sort(pangramChk); //as array has in-built method for sorting 
        String sorted = new String(pangramChk); 
        // converting array into string again as due to sorting all the chars are now arranged in increasing order of ascii values 
        sentence =  sorted.replaceAll("(.)\\1{1,}", "$1");
        //using regex to remove duplicates as due to sorting all the duplicates will be next to one another
        //replaceAll("regex"), "replacement" Replaces all occurances of "regex" with "replacement.
        //due to all this checks now if our string/sentence is pangram it must be having 26 unique letter/chars ie alphabets  
                     if(sentence.length()==26)
                      {
                       System.out.println("pangram");
                     }
                    else
                       {
                        System.out.println("not pangram");
                      }
                   }
                }

File: IsPangram/is_pangram.py
s = input().replace(' ','').lower() # input-constraints {a-z, A-Z, space}
letter_set = set(s)
print('pangram' if len(letter_set) == 26 else 'not pangram')


#--------------------------------------------------------------------------------------------------------------------------

from collections import Counter
s = input().lower()
s_char_set = Counter(s)
s_char_set.pop(' ', None)

print('pangram' if len(s_char_set) == 26 else 'not pangram')


#--------------------------------------------------------------------------------------------------------------------------

from collections import defaultdict

s = input().lower()
char_dict = defaultdict(lambda:0)

for char in s:
    char_dict[char] += 1

char_dict.pop(' ',None)

print('pangram' if len(char_dict) == 26 else 'not pangram')
